upf/upf_pfcp_api.c

```diff
@@ -803,16 +807,83 @@ lookup_nwi (u8 * name)
   return pool_elt_at_index (gtm->nwis, p[0]);
 }
 
+static u32
+generate_teid (u32 start, u32 mask)
+{
+  return random_u32(&start) & (UINT32_MAX ^ mask);
+}
```

Comment:
* that does not update the RND state and would therefore always generate the same TEID
* TEIDs 0 and 0xffffffff are reserved, need to check that we don't generate them